src/ModelFactory.php

```diff
@@ -18,11 +15,13 @@ public function __construct()
         parent::__construct(static::getClass());
     }
 
-    public static function __callStatic(string $name, array $arguments)
+    /**
+     * @return list<TModel&Proxy<TModel>>
+     * @psalm-return list<Proxy<TModel>>
+     */
+    public static function createMany(): array
```

Comment:
Shouldn't we use the real method parameters here? (`int $number, array|callable $attributes = []`)?

Reply:
Yes, you are correct. I've updated the PR.